rdrf/rdrf/filestorage.py

```diff
@@ -1,39 +1,93 @@
 # Attempt to clean up the interaction with GridFS
 import logging
+import re
+import gridfs
+from bson.objectid import ObjectId
+from .models import Registry, CDEFile
+from .utils import models_from_mongo_key
 
 logger = logging.getLogger("registry_log")
 
 
-class GridFSApi(object):
-    def __init__(self, gridfs_filestore):
-        self.gridfs_filestore = gridfs_filestore
-
-    # Public API
-
-    def store(self, registry_model, patient_model, file_cde_model, file_like_object):
-        """
-        Store a file uploaded by a user into gridfs - return the the gridfs file id
-        """
-        original_filename = file_like_object.name
-        added_file_id = self.gridfs_filestore.put(file_like_object.read(), filename=original_filename)
-        logger.info("Added file to gridfs for %s %s %s %s id = %s" % (registry_model.code,
-                                                                      patient_model,
-                                                                      file_cde_model.code,
-                                                                      original_filename,
-                                                                      added_file_id))
-        gridfs_ref_dict = {"gridfs_file_id": added_file_id, "file_name": original_filename}
-        logger.debug("gridfs_ref_dict = %s" % gridfs_ref_dict)
-        return gridfs_ref_dict
-
-    def delete(self, gridfs_file_id):
-        if self.gridfs_filestore.exists(gridfs_file_id):
-            self.gridfs_filestore.delete(gridfs_file_id)
-            logger.info("GridFS API: deleted %s OK" % gridfs_file_id)
-        else:
-            logger.info("%s did not exist in filestore")
+class FileStoreUtil(object):
```

Comment:
I can't see how polymorphism would make things better because the id type needs to be tested for each `file_ref` value.

Actually these methods shouldn't even be in a class.